### PR TITLE
Hide the customizer when the active theme is a FSE theme

### PR DESCRIPTION
**Pull request by albarin**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:
This PR makes sure that when a block theme is active, activating WooCommerce core will not enable the `Customizer`  option from the Appearance settings section and the `Customize` button from the WP admin

For more context see: https://woofse.wordpress.com/2022/06/07/pt-ensure-customizer-doesnt-show-up-with-block-themes-active-2/

### How to test the changes in this Pull Request:

1. Enable `WooCommerce`.
2. Enable the `Storefront` theme.
3. Check that the `Customize` button appears on the `Appearance` menu and on the menu bar.
4. Enable a block theme (i.e. `Twentytwentytwo`).
5. Check that the `Customize` button does not show up on the `Appearance` menu or on the menu bar.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php [hunk 1]
```diff
@@ -17,10 +17,12 @@ class WC_Shop_Customizer {
 	 * Constructor.
 	 */
 	public function __construct() {
-		add_action( 'customize_register', array( $this, 'add_sections' ) );
-		add_action( 'customize_controls_print_styles', array( $this, 'add_styles' ) );
-		add_action( 'customize_controls_print_scripts', array( $this, 'add_scripts' ), 30 );
-		add_action( 'wp_enqueue_scripts', array( $this, 'add_frontend_scripts' ) );
+		if ( ! wc_current_theme_is_fse_theme() ) {
```
Comment (barryhughes) [inline plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php]: Just a thought: if we literally don't want this instance to do anything, we could avoid instantiating it altogether (which would be slightly more efficient).

For instance, we could move this new condition to [the bottom of the file](https://github.com/woocommerce/woocommerce/blob/a48e7bb974614d5f3a335911e254342596069537/plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php#L890):

```php
if ( ! wc_current_theme_is_fse_theme() ) {
	new WC_Shop_Customizer();
}
```

(We don't keep any references to the object, so this should be equally safe.)

wdyt?

**Review by barryhughes — CHANGES_REQUESTED** [review]

File: plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php (hunk 1 above)
Comment (albarin) [reply plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php]: Yep, makes total sense 😅 
Any suggestions about how to go about testing this in that case? I guess with this new approach the tests on [class-wc-shop-customizer-test.php](https://github.com/woocommerce/woocommerce/blob/a48e7bb974614d5f3a335911e254342596069537/plugins/woocommerce/tests/php/includes/customizer/class-wc-shop-customizer-test.php) don't make much sense anymore. 🤔

Comment (albarin) [reply plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php]: @barryhughes Also, I see a couple of checks failing that seem unrelated to the changes (I saw the same failures on other PRs), any idea?

Comment (barryhughes) [reply plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php]: > Any suggestions about how to go about testing this in that case?

Yes, it's a little awkward and I think any tests would be brittle. Given the nature of the change and the fact we're working with some legacy structures and conventions, I'd tend to think we can skip adding tests for this.

> Also, I see a couple of checks failing that seem unrelated to the changes

Yep, we can probably ignore. I'll rebase though to see if we can clear the failing tests under PHP 7.4 specifically.

Comment (albarin) [reply plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php]: Makes sense! Then I think we are good to go 😄 Thanks a lot for your help! 🙌

**Review by barryhughes — APPROVED** [review]

**Pull request merged**